### DISCUSSION TITLE
Is arm64 supported?

### DISCUSSION THREAD

**Discussion by Levi-Armstrong** [open-telemetry/opentelemetry-cpp#2261] [General]
I noticed that some of the other repos in this organization has arm64 CI build but this one does not. Is there any reason this would not work on arm64?

**Comment by lalitb** [comment]
@Levi-Armstrong ARM64 is definitely supported. We just don't have the CI. There is a open issue for same - https://github.com/open-telemetry/opentelemetry-cpp/issues/1417 , if someone wants to volunteer adding it.

**Reply by Levi-Armstrong** [reply]
Thank you for the quick response and will post any issue we have if any trying to build on arm64.

**Comment by esigo** [comment]
The reason we don't have a CI for arm64, is that it takes really long time to build. We would need a registry to push images for dependencies e.g. `gRPC` for arm64 builds, to make the builds faster, but currently we don't have that. I was trying to use GitHub artifacts for building arm64 using `buildx` but it turned it's not possible to save arm64 builds on x64.

**Comment by lalitb** [comment]
Just a thought, we can at-least add CI to build the otel-cpp for ARM using clang/gcc cross-compiler. Mayn't run the unit-tests as that would need the qemu emulator and could be slow. There are various container offering for ARM cross compiler on x86 containers to use it - https://github.com/dockcross/dockcross. Sort of partial test, but good to investigate.
